Automattic/jetpack
# jetpack: Use Webpack to build modules JS

#### Changes proposed in this Pull Request:

Currently the build uses an old version of Terser via gulp-minify to
simply minify the various modules. Let's switch that over to Webpack
where we have Babel for transpilation, a newer version of Terser, and so
on.

This did necessitate a few changes to some of the existing modules:
* Adjust a few bundled libraries' comments to preserve licenses that
  were being preserved previously.
* Adjust `modules/likes/post-count.js` and `post-count-jetpack.js` to
  preserve the `wpPostLikeCount` global that it seems like was there
  intentionally.
* Adjust `modules/widgets/google-translate/google-translate.js` so the
  global function is still global.

~~A followup will then rename these back from ".min.js" and use
`Assets::register_script()` to register the scripts.~~

~~There's also now an opportunity here for someone to update the modules
to modern JS syntax (now that we have Babel in the pipeline), to pull in
libraries like swiper via package.json instead of bundling them manually
(now that we're using an actual bundler), and maybe even to use wp.i18n.
I'll leave all that for someone else to do.~~

Unfortunately further improvements like those mentioned above will have to
wait until wpcom uses Jetpack rather than Fusion.

#### Jetpack product discussion


PT: p9dueE-3MG-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Is the build happy?
* Check that modules in general still work. Pay special attention to the following:
  * Carousel, swiper still works right.
  * Loss of global variables (global state pollution?) from `modules/likes/queuehandler.js` is ok.
    * I went out of my way to preserve a "wpPostLikeCount" on `window` from `modules/likes/post-count.js`. I could revert that if I was mistaken in thinking that's needed.
  * Loss of global `JetpackSlideshow` (global state pollution?) from `modules/shortcodes/js/slideshow-shortcode.js` is ok.
  * Loss of global `Milestone` (global state pollution?) from `modules/widgets/milestone/milestone.js` is ok.
  * Loss of global `jetpackSearchModule` (global state pollution?) from `modules/widgets/search/js/search-widget.js` is ok.
  * I tried to preserve the behavior of `modules/widgets/google-translate/google-translate.js`. Did I succeed?
